src/qibolab/instruments/erasynth.py

```diff
@@ -14,18 +14,17 @@
 
 
 class ERA(LocalOscillator):
-    def __init__(self, name, address, ethernet=True):
+    def __init__(self, name, address, ethernet=True, reference_clock_source="internal"):
         super().__init__(name, address)
         self.device: ERASynthPlusPlus = None
         self._power: int = None
         self._frequency: int = None
         self.ethernet = ethernet
         self._device_parameters = {}
+        self.reference_clock_source = reference_clock_source
 
     @property
     def frequency(self):
-        if self.is_connected:
-            return self.device.get("frequency")
```

Comment:
I understand that this does not work because in some cases `device` is `None` (depending on how we are connecting to it), however keep in mind that ideally we would like to get the frequency (and power) value from the device if we do not set it. For example the following
```py
from qibolab.instruments.erasynth import ERA

era = ERA(f"era_01", f"192.168.0.201")
print(era.frequency) # it is None because we have not set it
era.connect()
print(era.frequency) # will still print None ignoring the value that is already set in the device
```

Although I have not tested it
```py
from qibolab.instruments.erasynth import ERA

era = ERA(f"era_01", f"192.168.0.201")
era.connect()
era.setup()
```
may also fail because `setup` will try to set `None` values as frequencies.

I think to properly fix the getters for all cases we need to implement an http `_get` method (symmetric to `_post`), but I am not sure if that works for that device.

Reply:
There is not a single driver that interrogates the instrument to get one of its parameter's values. When you do `self.device.get("frequency")`, you are calling the `qcodes` driver which will return the value that's saved in its data structure. It is then equivocal to directly return `_value` in our data structure.

Reply:
Well I just tested for Rohde Schwarz and
```py
from qibolab.instruments.rohde_schwarz import SGS100A

lo = SGS100A("lo", "192.168.0.34")
print(lo.frequency) # None because we have not set it
lo.connect()
print(lo.frequency) # prints 7.1GHz which was the frequency the instrument already had
```
so it seems that it is actually "interrogating" the instrument.

Reply:
Ok fair enough. Note that it is not the case for the SPI. 
As it is not so hard to do, I've done it